File: src/main/java/com/taxidriverhk/hkadbus2/util/IdentifierGenerator.java
package com.taxidriverhk.hkadbus2.util;

import java.math.BigInteger;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class IdentifierGenerator {

    public static Long generateNumericIdentifier(final String uuid) {
        final String uuidWithoutDashes = uuid.replaceAll("-", "");
        // UUID without dashes is a number in hexadecimal format
        final BigInteger bigNumber = new BigInteger(uuidWithoutDashes, 16);

        final String numericIdString = bigNumber.toString();
        final int digits = numericIdString.length();
        // Obtain the last 18 digits
        final String shortenNumericId = numericIdString.substring(digits - 18, digits);

        return Long.valueOf(shortenNumericId);
    }
}

File: src/main/kotlin/com/taxidriverhk/hkadbus2/util/IdentifierGenerator.kt
package com.taxidriverhk.hkadbus2.util

import java.math.BigInteger

object IdentifierGenerator {

    @JvmStatic
    fun generateNumericIdentifier(uuid: String): Long {
        val uuidWithoutDashes = uuid.replace("-", "")
        val bigNumber = BigInteger(uuidWithoutDashes, 16)

        val numericIdString = bigNumber.toString()
        val digits = numericIdString.length
        val shorternNumericId = numericIdString.substring(digits - 9, digits)

        return shorternNumericId.toLong()
    }
}
